### PR TITLE
fix(router-plugin): trigger navigation on the queryparams change

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Issue Number: #922 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/router-plugin/src/router.state.ts [hunk 1]
```diff
@@ -107,12 +113,8 @@ export class RouterState {
     });
   }
```
Comment (splincode) [inline packages/router-plugin/src/router.state.ts]:
```ts
this._router.events.subscribe(e => {
  if (RouterState.checkRouterEvent(e, [RoutesRecognized])) {
    ..
  } else if (RouterState.checkRouterEvent(e, [GuardsCheckEnd, ResolveEnd])) {
  ..
  } else if (RouterState.checkRouterEvent(e, [NavigationEnd]) {
  ..
  }
})

private static checkRouterEvent(e: RouterEvent, eventTypes: RouterEvent[]) {
  return events.some(event => e incenceof event);
}
```

Comment (splincode) [reply packages/router-plugin/src/router.state.ts]: what do you mean?

Comment (markwhitfeld) [reply packages/router-plugin/src/router.state.ts]: @splincode I think that your suggested code makes things more complicated for not much gain.

Comment (markwhitfeld) [reply packages/router-plugin/src/router.state.ts]: @splincode The change you are suggesting makes things more complicated for very little benefit. It doesn't reduce repetition either. The existing code is easy to understand and can be optimised by the JS engine too.

**Pull request merged**